packages/multichain-account-service/src/MultichainAccountService.ts

```diff
@@ -395,52 +389,187 @@ export class MultichainAccountService {
+   * @returns The new multichain account wallet.
+   */
+  async #createWalletByNewVault(
+    password: string,
+  ): Promise<MultichainAccountWallet<Bip44Account<KeyringAccount>>> {
+    log(`Creating new wallet by creating a new vault and keychain...`);
+    await this.#messenger.call(
+      'KeyringController:createNewVaultAndKeychain',
+      password,
+    );
+
+    const entropySourceId = (await this.#messenger.call(
+      'KeyringController:withKeyring',
+      { type: KeyringTypes.hd },
+      async ({ metadata }) => {
+        return metadata.id;
+      },
+    )) as string;
+
+    // The wallet is ripe for discovery
+    return new MultichainAccountWallet({
+      providers: this.#providers,
+      entropySource: entropySourceId,
+      messenger: this.#messenger,
+    });
+  }
+
+  /**
+   * Creates a new multichain account wallet by restoring a vault and keyring.
+   *
+   * @param password - The password to encrypt the vault with.
+   * @param mnemonic - The mnemonic to use to restore the new wallet.
+   * @returns The new multichain account wallet.
+   */
+  async #createWalletByRestore(
+    password: string,
+    mnemonic: Uint8Array,
+  ): Promise<MultichainAccountWallet<Bip44Account<KeyringAccount>>> {
+    log(`Creating new wallet by restoring vault and keyring...`);
+    await this.#messenger.call(
+      'KeyringController:createNewVaultAndRestore',
+      password,
+      mnemonic,
+    );
+
+    const entropySourceId = (await this.#messenger.call(
+      'KeyringController:withKeyring',
+      { type: KeyringTypes.hd },
+      async ({ metadata }) => {
+        return metadata.id;
+      },
+    )) as string;
+
+    // The wallet is ripe for discovery
+    return new MultichainAccountWallet({
+      providers: this.#providers,
+      entropySource: entropySourceId,
+      messenger: this.#messenger,
+    });
+  }
+
+  /**
+   * Creates a new multichain account wallet by either importing an existing mnemonic,
+   * creating a new vault and keychain, or restoring a vault and keyring.
+   *
+   * NOTE: This method should only be called in client code where a mutex lock is acquired.
+   * `discoverAccounts` should be called after this method to discover and create accounts.
+   *
+   * @param options - Options.
+   * @param options.mnemonic - The mnemonic to use to create the new wallet.
+   * @param options.password - The password to encrypt the vault with.
+   * @param options.type - The flow type to use to create the new wallet.
+   * @throws If the mnemonic has already been imported.
+   * @returns The new multichain account wallet.
+   */
+  async createMultichainAccountWallet({
+    mnemonic,
+    password,
+    type,
+  }: CreateWalletParams): Promise<
+    MultichainAccountWallet<Bip44Account<KeyringAccount>>
+  > {
+    const params: CreateWalletValidatedParams =
+      this.#getValidatedCreateWalletParams({
+        mnemonic,
+        password,
+        type,
+      });
```

Comment:
I think you can simplify this with:

```suggestion
  async createMultichainAccountWallet(params: CreateWalletParams): Promise<
    MultichainAccountWallet<Bip44Account<KeyringAccount>>
  > {
```

The union should naturally be inferred thanks to your `if (params.type ===` below

Which also means we can remove `#getValidatedCreateWalletParams` entirely too!